Lib/typing.py

```diff
@@ -1675,13 +1675,66 @@ class _TypingEllipsis:
 
 _SPECIAL_NAMES = frozenset({
     '__abstractmethods__', '__annotations__', '__dict__', '__doc__',
-    '__init__', '__module__', '__new__', '__slots__',
-    '__subclasshook__', '__weakref__', '__class_getitem__',
-    '__match_args__',
+    '__module__', '__slots__', '__match_args__', '__qualname__',
+})
+_SPECIAL_CALLABLE_NAMES = frozenset({
+    '__init__', '__new__', '__subclasshook__','__class_getitem__', '__weakref__',
 })
 
 # These special attributes will be not collected as protocol members.
 EXCLUDED_ATTRIBUTES = _TYPING_INTERNALS | _SPECIAL_NAMES | {'_MutableMapping__marker'}
+EXCLUDED_MEMBERS = EXCLUDED_ATTRIBUTES | _SPECIAL_CALLABLE_NAMES
```

Comment:
Please keep the diff as minimal as possible. As far as I can tell, splitting the `_SPECIAL_NAMES` frozenset into a `_SPECIAL_NAMES` set and a `_SPECIAL_CALLABLE_NAMES` set shouldn't make any difference here

Reply:
For my solution to work, I need both, because for things defined directly inside the protocol body, I only exclude `EXCLUDED_ATTRIBUTES`, but not `_SPECIAL_CALLABLE_NAMES`. (Hence, we can test for `__class_getitem__`). However, for general classes we need to exclude it (otherwise `test.test_typing.ProtocolTests.test_collections_protocols_allowed` fails).

See the difference in function body between `_get_local_members` and `_get_local_protocol_members`.